SYCL/UserDefinedReductions/user_defined_reductions.cpp

```diff
@@ -0,0 +1,71 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
+
+#include <numeric>
+
+#include <sycl/ext/oneapi/experimental/user_defined_reductions.hpp>
+#include <sycl/sycl.hpp>
+
+template <typename T> struct UserDefinedSum {
+  T operator()(T a, T b) { return a + b; }
+};
```

Comment:
I think we need to test different op/identity/default-init value (e.g., unequal to identity/deleted default ctor if allowed).

Reply:
Added in https://github.com/intel/llvm-test-suite/pull/1395/commits/760338ccf459ff8a64aae77a94ded0fc7c30cb7c